lib/deliver/submit_for_review.rb

```diff
@@ -29,10 +29,22 @@ def submit!(options)
     end
 
     def select_build(options)
-      Helper.log.info "Selecting the latest build..."
       app = options[:app]
       v = app.edit_version
-      build = wait_for_build(app)
+
+      if options[:build_number]
+        Helper.log.info "Selecting existing build-number: #{options[:build_number]}"
+        build = v.candidate_builds.detect { |a| a.build_version == options[:build_number] }
+        unless build
```

Comment:
Ugh, that seems kind of dirty, we first try to find a build, and then check if it's `"latest"`? How about we check it before and change the first `if` to

```ruby
if options[:build_number] and options[:build_number] != "latest"
```